<!-- Dataset Frankfurt Main Tower from TU Chemnitz smartLoc project. -->
<launch>
    <!-- load dataset-specific params -->
    <rosparam command="load" file="$(find gnss_preprocessor)/config/frankfurt.yaml" />

    <!-- path of dataset -->
    <param name="roverMeasureFile" type="string" value="$(find gnss_preprocessor)/dataset/smartLoc/Frankfurt_Main_Tower/Frankfurt_Main_Tower.obs" />
    <param name="EmpFile" type="string" value="$(find gnss_preprocessor)/dataset/smartLoc/Frankfurt_Main_Tower/BRDC00WRD_R_20162520000_01D_MN.rnx" />
<!--
    <param name="GLONASSEmpFile" type="string" value="$(find gnss_preprocessor)/dataset/smartLoc/Frankfurt_Main_Tower/brdc2520.16g" />
    <param name="GalileoEmpFile" type="string" value="$(find gnss_preprocessor)/dataset/smartLoc/Frankfurt_Main_Tower/wroc2520.16l" />
    <param name="GPSEmpFile" type="string" value="$(find gnss_preprocessor)/dataset/smartLoc/Frankfurt_Main_Tower/brdc2520.16n" />
-->
    <param name="SP3File" type="string" value="$(find gnss_preprocessor)/dataset/smartLoc/Frankfurt_Main_Tower/gbm19134.sp3" />
    <param name="clkFile" type="string" value="$(find gnss_preprocessor)/dataset/smartLoc/Frankfurt_Main_Tower/gbm19134.clk" />
    <param name="SBASFile" type="string" value="$(find gnss_preprocessor)/dataset/smartLoc/Frankfurt_Main_Tower/REDU00BEL_R_20162520000_01D_SN.rnx" />
    <param name="ionexFile" type="string" value="$(find gnss_preprocessor)/dataset/smartLoc/Frankfurt_Main_Tower/codg2520.16i" />
    <param name="atxFile" type="string" value="$(find gnss_preprocessor)/dataset/Shared/igs14.atx" />

    <param name="out_folder"      type="string" value="$(find gnss_preprocessor)/dataset/smartLoc/Frankfurt_Main_Tower/rtklibResult.pos" />

    <node name="gnss_preprocessor_node" pkg="gnss_preprocessor" type="gnss_preprocessor_node" output="screen" />
    

    <!-- Static fake TF transform -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_tf_fake" args="0 0 0 0 0 0 map base_link" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_tf_fake_m2w" args="0 0 0 0 0 0 map world" />

    <!-- Rosbag recording -->
    <node pkg="rosbag" type="record" name="rosbag_Frankfurt_Main_Tower" args="record -O $(find gnss_preprocessor)/dataset/Processed/smartLoc_Frankfurt_Main_Tower_preprocessed.bag -a"/> 

</launch>
